plugin/src/test/java/io/jenkins/plugins/coverage/model/DockerGitForensicsITest.java

```diff
@@ -0,0 +1,294 @@
+        setCoveragePublisherForFreeStyleProject(project, JACOCO_REFERENCE_FILE);
+        Run<?, ?> referenceBuild = buildSuccessfully(project);
+
+        setGitScmWithCommitSpecForFreeStyleProject(project, COMMIT);
+        setCoveragePublisherForFreeStyleProject(project, JACOCO_FILE);
+        Run<?, ?> build = buildSuccessfully(project);
+
+        verifyGitIntegration(build, referenceBuild);
+    }
+
+    /**
+     * Verifies the Git repository for the commit with the passed ID.
+     *
+     * @param build
+     *         The current build
+     * @param commit
+     *         The commit ID
+     */
+    private void verifyGitRepositoryForCommit(final Run<?, ?> build, final String commit) {
+        String consoleLog = getConsoleLog(build);
+        assertThat(consoleLog)
+                .contains("remote.origin.url " + REPOSITORY)
+                .contains("Checking out Revision " + commit)
+                .contains("checkout -f " + commit);
+    }
+
+    /**
+     * Verifies the Git integration.
+     *
+     * @param build
+     *         The current build
+     * @param referenceBuild
+     *         The reference build
+     */
+    private void verifyGitIntegration(final Run<?, ?> build, final Run<?, ?> referenceBuild) {
+        CoverageBuildAction action = build.getAction(CoverageBuildAction.class);
+        assertThat(action).isNotNull();
+        assertThat(action.getReferenceBuild())
+                .isPresent()
+                .satisfies(reference ->
+                        assertThat(reference.get().getExternalizableId()).isEqualTo(
+                                referenceBuild.getExternalizableId()));
+        verifyCodeDelta(action);
+        verifyCoverage(action);
+    }
+
+    /**
+     * Verifies the calculated coverage for the most important metrics line and branch coverage.
+     *
+     * @param action
+     *         The created Jenkins action
+     */
+    private void verifyCoverage(final CoverageBuildAction action) {
+        verifyOverallCoverage(action);
+        verifyChangeCoverage(action);
+        verifyIndirectCoverageChanges(action);
+    }
+
+    /**
+     * Verifies the calculated overall coverage including the coverage delta.
+     *
+     * @param action
+     *         The created Jenkins action
+     */
+    private void verifyOverallCoverage(final CoverageBuildAction action) {
+        assertThat(action.getLineCoverage()).satisfies(coverage -> {
+            assertThat(coverage.getCovered()).isEqualTo(546);
+            assertThat(coverage.getMissed()).isEqualTo(461);
+        });
+        assertThat(action.getBranchCoverage()).satisfies(coverage -> {
+            assertThat(coverage.getCovered()).isEqualTo(136);
+            assertThat(coverage.getMissed()).isEqualTo(94);
+        });
+        assertThat(action.getDifference()).contains(
+                new SimpleEntry<>(LINE, CoveragePercentage.getCoveragePercentage(65_160, 103_721)),
```

Comment:
maybe rename the simple factory method to `valueOf`?
```suggestion
                new SimpleEntry<>(LINE, CoveragePercentage.valueOf(65_160, 103_721)),
```